gradle-plugin/src/main/kotlin/io/spine/protodata/gradle/Plugin.kt

```diff
@@ -165,7 +181,24 @@ private fun Project.configureProtobufPlugin(extension: Extension, version: Strin
                 project.tasks.getByName(launchTaskName(it.sourceSet)).dependsOn(it)
             }
         }
+        generatedFilesBaseDir = "$buildDir/generated-proto/"
     }
 
 private fun launchTaskName(sourceSet: SourceSet): String =
     "launchProtoData${sourceSet.name.capitalize()}"
+
+
+private fun Project.configureSourceSets(extension: Extension) {
+    afterEvaluate {
+        sourceSets.forEach { sourceSet ->
+            sourceSet.java.srcDir(extension.targetDir(sourceSet))
+
+            val sourceDir = file(extension.sourceDir(sourceSet))
+            val task = javaCompileFor(sourceSet)!!
+            task.source = task.source.filter { file -> !file.liesInside(sourceDir) }.asFileTree
+        }
+    }
+}
+
+private fun File.liesInside(directory: File): Boolean =
```

Comment:
I would use "laysInside" to avoid confusion. It's technically the same, but at the same time, it looks truthfully.

Or we could go for "residesIn"?

Reply:
Reconsidering the "laysInside" option — it's not quite good for this case.

Let's go with something like `residesIn`.